### ISSUE TITLE
The use of CustomCentroidBondForce as a BiasVariable in a metadynamics simulation of a protein-protein dissociation

### ISSUE THREAD

**Issue by mjw99** [openmm/openmm#4589]
I have been looking at the new metadynamics functionality in OpenMM 8.1.1 and I am trying to reproduce a protocol in the following [paper](https://www.nature.com/articles/s41598-022-05875-8#Sec11):

_....a complex is dissociated by performing a standard metadynamics run with a user-defined distance between the two molecules used as the collective variable (CV). A Gaussian energy impulse centered at a running value of the CV is periodically added to the total potential energy of the system at every time interval along the progression of the simulation, which forces the CV to increase, leading to the dissociation of the molecules..._

I prepare one of the systems (3F1P: HIF2A–ARNT) using a standard NPT workflow with an explicit solvent and this is all stable. Next I specifically setup the metadynamics with the following, attempting to match the quoted metadynamics parameters paper, as close as possible:

```
# For example PDB 3F1P from https://www.nature.com/articles/s41598-022-05875-8/tables/1
# Create the CV: V340A–I458B

#V340A
val_index = [atom.index for atom in pdb.topology.atoms() if (atom.residue.id in ['340'] and atom.residue.chain.id == 'A' and atom.name in ['N','CA','C','O'])]
#I458B
iso_index = [atom.index for atom in pdb.topology.atoms() if (atom.residue.id in ['458'] and atom.residue.chain.id == 'B' and atom.name in ['N','CA','C','O'])]


force = mm.CustomCentroidBondForce(2, "distance(g1, g2)")

force.addGroup(val_index)
force.addGroup(iso_index)

force.addBond([0, 1])
force.setUsesPeriodicBoundaryConditions(True)


V340AI458B = app.BiasVariable(force=force,
                              minValue=6 * unit.angstrom,
                              maxValue=30 * unit.angstrom,
                              biasWidth=0.05 * unit.angstrom,
                              periodic=False)


meta = app.Metadynamics(system=system,
                        variables=[V340AI458B],
                        temperature=310.0,
                        biasFactor=5.0,
                        height=0.01 * unit.kilocalories_per_mole,
                        frequency=45)
....

meta.step(simulation, total_steps)
``` 
My issue is that over ~40 ns, via visual inspection, the proteins do not dissociate. 

From the initial overview above, is there anything obvious that I have missed? Is the form of my potential in the `CustomCentroidBondForce` constructor wrong and I should be using the full harmonic form and somehow be providing an equilibrium distance parameter via addCollectiveVariable()? Or, perhaps, should I be making use of `pointdistance(x1, y1, z1, x2, y2, z2)` in some form.  The full example can be found [here](https://github.com/mjw99/openmm_meta).

As a side point, I can reproduce the alanine dipeptide example from Peter, all fine and this is also in the that repo too.

Thanks

Mark

**Comment by peastman** [comment]
I don't see anything obviously wrong.  What is the distance between the groups?  Make sure it's really in the range 6 to 30 Å.  You can also take a look at `meta._totalBias`, which contains the bias being added to the energy (in kJ/mol).  If the distance isn't increasing, you should see a lot of bias accumulating at that distance.

**Comment by mjw99** [comment]
Dear Peter,

Thank you for looking over this; it is appreciated. 

I've managed to tweak, mainly the BiasVariable parameters (https://github.com/mjw99/openmm_meta/commit/3e39b90a92b0fac84b875d1500e96f2cad66761b), to obtain a shorted, minimal, example whereby this works ~50% of the time. I've also plotted the distances between the two CA residues involved in the CustomCentroidBondForce as a convenient metric of success. These are shown in the repo as dist_{good,bad}.png.

I am happy with this so far, as it indicates that the metadynamics is doing as instructed, but I am still trying to grok the stochastic nature of this result. I also assume if one's system has a timestep of 2 fs and the Metadynamics(..., frequency=2, ...), then the injection of the Gaussian will be every 4 fs?

**Comment by peastman** [comment]
> I also assume if one's system has a timestep of 2 fs and the Metadynamics(..., frequency=2, ...), then the injection of the Gaussian will be every 4 fs?

Correct, though you don't really want to do it nearly that frequently.

**Comment by qweiliang1** [comment]
Hi @mjw99,
Thanks for publishing the example of your metadyanmics run. It has been most useful for me in getting my own metadynamics run going.

Dear @peastman, 
May I ask something about the outputted bias from the Metadynamics class? Specifically in the example from @mjw99, the biases outputs a np array of length 1200. What do the values represent, and what affects the length of the output array?
